luigi_pipeline/seqr_mito_loading.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import sys
+
+import luigi
+import hail as hl
+
+from lib.model.mito_mt_schema import SeqrMitoVariantsAndGenotypesSchema, SeqrMitoVariantSchema, SeqrMitoGenotypesSchema
+from luigi_pipeline.seqr_loading_optimized import BaseVCFToGenotypesMTTask, BaseMTToESOptimizedTask
+from seqr_loading import SeqrVCFToMTTask
+
+logger = logging.getLogger(__name__)
+
+
+class SeqrMitoVariantMTTask(SeqrVCFToMTTask):
+    """
+    Loads all annotations for the variants of a Matrix Table into a Hail Table.
+    """
+    high_constraint_interval_path = luigi.Parameter(description='Path to the tsv file storing the high constraint intervals.')
+    RUN_VEP = False
+    SCHEMA_CLASS = SeqrMitoVariantSchema
+
+    def get_schema_class_kwargs(self):
+        kwargs = super().get_schema_class_kwargs()
+        kwargs['high_constraint_region'] = hl.import_locus_intervals(self.high_constraint_interval_path,
+                                                                     reference_genome='GRCh38')
+        return kwargs
+
+    def import_dataset(self):
+        mt = hl.read_matrix_table(self.source_paths[0])
+        return mt.drop('dbsnp_version',
```

Comment:
why is this necessary? The annotation step should ignore unused fields and we then call `select_annotated_mt` which selects only the needed rows for export

Reply:
The `select_annotated_mt` doesn't select the global fields. I can add a selection for them. Then, we don't need to drop extra fields here.

Reply:
It is hard to select the global fields when calling the `select_annotated_mt` because we don't know which of them we need. Maybe, it is better for us to select them we need here.

Reply:
It is better for us to share code with the SNP pipeline, which successfully outputs all the fields it needs and no more. We export globals just fine for SNPs so I don't see why that would be a problem here. Figure out a way so that the existing call of `self.SCHEMA_CLASS(mt, **kwargs).annotate_all(overwrite=True).select_annotated_mt()` handles all of the field selection we need. If we need to make modifications to what we are annotating or what we are selecting, make those changes inside the helper functions that do the annotation or selection, not here. Remeber, partially overriding `annotate_all` or `select_annotated_mt` or adding additional conditional logic in those functions is allowed

Reply:
> It is hard to select the global fields when calling the select_annotated_mt because we don't know which of them we need. 

We absolutely do know what fields we need in the outputted index, you wrote a detailed document listing all of them. Moreover, for SNPs (and now this pipeline since you are using shared code) the call the `annotate_globals` is run after the call to `select_annotated_mt` so the necessary globals are explicitly set rather than accidentally being carried over from the input. This is the behavior we want for mito. If there are any additional globals you think we need for mito, we should add a helper function that generates the globals for an index and override it for mito

Reply:
There are no additional global fields for mito. So, it is weird to override `select_annotated_mt` for mito. I also found that the `_meta` fields in the existing Elasticsearch indices contain only those fields explicitly annotated in the pipeline. The `select_globals` method can do annotation and selection of the global fields at the same time.  How about we changing `annotate_globals` to `select_globals`?

Reply:
I don't understand why anything needs to change at all

Reply:
sorry github was showing this comment really weirdly that mad it hard to see what you had actually changed, this looks fine now